src/librustc_codegen_ironox/value.rs

```diff
@@ -21,9 +21,11 @@ pub enum Value {
     /// local value.
     Local(usize, usize),
     /// An unspecified constant. This is just a wrapper around a `Type`.
-    ConstUndef,
-    BigConst(u128),
-    Global,
+    ConstUndef(Type),
```

Comment:
I'm intrigued. Where would an unspecified constant arise?

Reply:
As far as I am aware, this is very similar to the 'undef' values you can create in LLVM. Essentially, a `ConstUndef` is an uninitialized constant of an arbitrary type. The backend has to implement a `const_undef` function which returns an uninitialized `Value`.

There are several places where this is used. The backend-agnostic code from `librustc_codegen_ssa` calls `const_undef` to create unitialized structs. The structs can later be  populated with values by calling `insert_value(struct, element, index)`.

Reply:
Ah ok. "Uninitialised" makes complete sense. I was confused by "undefined", which is a scary word in the land of systems programming.